compilation error when passing option type to a function

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.1.29 9fdf04b.e250547
**OS:** Windows 7 Professional (x64, little-endian)
**CC version:** clang version 10.0.0 (https://github.com/llvm/llvm-project.git d32170dbd5b0d54436537b6b75beaf44324e0c28)

**What did you do?**
```v
fn optional_arg(x ?int) {
	println("int type")

}

fn main() {
	optional_arg(1)
}
```


**What did you expect to see?**
I expected V to either to throw an error for unable to convert between `any_int` to `option_int` or convert the corresponding type to its option counterpart
 
**What did you see instead?**
```
C:\Users\lambda\AppData\Local\Temp\v\hello_world.1829071214341309535.tmp.c:8082:21: error: passing 'int' to parameter of incompatible type 'Option_int' (aka 'struct Option_int')
        main__optional_arg(1);
                           ^
C:\Users\lambda\AppData\Local\Temp\v\hello_world.1829071214341309535.tmp.c:8077:36: note: passing argument to parameter 'x' here
void main__optional_arg(Option_int x) {
                                   ^
9 warnings and 1 error generated.
```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -181,6 +181,9 @@ fn (mut c Checker) fn_decl(mut node ast.FnDecl) {
 				c.error('invalid use of reserved type `$param.name` as a parameter name',
 					param.pos)
 			}
+			if param.typ.has_flag(.optional) {
+				c.error('optional type argument is not supported currently', param.type_pos)
+			}
 			if !param.typ.is_ptr() { // value parameter, i.e. on stack - check for `[heap]`
 				arg_typ_sym := c.table.sym(param.typ)
 				if arg_typ_sym.kind == .struct_ {

--- a/vlib/v/checker/tests/fn_arg_of_optional_err.out
+++ b/vlib/v/checker/tests/fn_arg_of_optional_err.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/fn_arg_of_optional_err.vv:1:19: error: optional type argument is not supported currently
+    1 | fn optional_arg(x ?int) {
+      |                   ^
+    2 |     println('int type: $x')
+    3 | }

--- a/vlib/v/checker/tests/fn_arg_of_optional_err.vv
+++ b/vlib/v/checker/tests/fn_arg_of_optional_err.vv
@@ -1,0 +1,7 @@
+fn optional_arg(x ?int) {
+	println('int type: $x')
+}
+
+fn main() {
+	optional_arg(1)
+}

